mteb/models/jina_models.py

```diff
@@ -0,0 +1,127 @@
+from __future__ import annotations
+
+import logging
+from collections.abc import Sequence
+from functools import partial
+from typing import Any
+
+import numpy as np
+import torch
+from sentence_transformers import __version__ as st_version
+
+from mteb.model_meta import ModelMeta
+
+from ..encoder_interface import PromptType
+from .sentence_transformer_wrapper import SentenceTransformerWrapper, get_prompt_name
+
+logger = logging.getLogger(__name__)
+
+MIN_SENTENCE_TRANSFORMERS_VERSION = (3, 1, 0)
+CURRENT_SENTENCE_TRANSFORMERS_VERSION = tuple(map(int, st_version.split(".")))
+
+
+class JinaWrapper(SentenceTransformerWrapper):
+    """following the hf model card documentation."""
+
+    jina_task_to_prompt = {
+        "retrieval.query": "Represent the query for retrieving evidence documents: ",
+        "retrieval.passage": "Represent the document for retrieval: ",
+    }
+
+    def __init__(
+        self,
+        model: str,
+        revision: str | None = None,
+        model_prompts: dict[str, str] | None = None,
+        **kwargs,
+    ) -> None:
+        if CURRENT_SENTENCE_TRANSFORMERS_VERSION < MIN_SENTENCE_TRANSFORMERS_VERSION:
+            raise RuntimeError(
+                f"sentence_transformers version {st_version} is lower than the required version 3.1.0"
+            )
+        try:
+            import einops  # noqa: F401
+        except ImportError:
+            raise ImportError(
+                "To use the jina-embeddings-v3 models `einops` is required. Please install it with `pip install einops`."
+            )
```

Comment:
just a small feedback, install flash attention will make evaluation much faster, maybe it's worth to give a logging message

Reply:
I would set it as a requirement.

I would also specify:

```pip install mteb[jina]```
for installing dependencies to ensure that the required dependencies are specified.

Reply:
Do you mean to include both `flash attention` and `einops`? I have some concerns about `flash attention` because it needs to be installed with `pip install flash-attn --no-build-isolation`, and it also requires CUDA 12.1

Reply:
Ahh, yeah, right. Then I would probably throw an info message for it and separate it out to its own optional dependency.